### PR TITLE
Dataflow SpannerIO sample

### PR DESCRIPTION
**Pull request by davidcavazos**
R: @jabubake 

This is a sample to show how to read from SpannerIO. In interest of time, I'll add the test in a separate PR.

### CODE REVIEW

File: dataflow/spanner-io/pom.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>proto-google-common-protos</artifactId>
+      <version>0.1.9</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-spanner</artifactId>
+      <version>0.20.0-beta</version>
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply dataflow/spanner-io/pom.xml]: Due to the grpc mismatch, had to leave it as `0.20-beta`

File: dataflow/spanner-io/src/main/java/com/example/dataflow/Spanner.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
+/*
+  Copyright 2016, Google, Inc.
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply dataflow/spanner-io/src/main/java/com/example/dataflow/Spanner.java]: Love the regex! Thanks (good old copy-paste)

File: dataflow/spanner-io/src/main/java/com/example/dataflow/Spanner.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    String getTableName();
+    void setTableName(String value);
+
+    @Description("Output filename for records size")
+    @Validation.Required
+    String getOutput();
+    void setOutput(String value);
+  }
+
+  public static class EstimateStructSizeFn extends DoFn<Struct, Long> {
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply dataflow/spanner-io/src/main/java/com/example/dataflow/Spanner.java]: Done

File: dataflow/spanner-io/src/main/java/com/example/dataflow/Spanner.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            throw new IllegalArgumentException("Arrays are not supported :(");
+          case STRUCT:
+            throw new IllegalArgumentException("Structs are not supported :(");
+        }
+      }
+      c.output(sum);
+    }
+  }
+
+  public static void main(String[] args) {
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply dataflow/spanner-io/src/main/java/com/example/dataflow/Spanner.java]: Added comments

File: dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerRead.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+  public static void main(String[] args) {
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+    Pipeline p = Pipeline.create(options);
+
+    String instanceId = options.getInstanceId();
+    String databaseId = options.getDatabaseId();
+    String query = "SELECT * FROM " + options.getTable();
+
+    PCollection<Long> tableEstimatedSize = p
```
Comment (mairbek) [inline dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerRead.java]: should we also create a sample that uses Read API?

**Pull request merged**